schema/derive/src/lib.rs

```diff
@@ -62,34 +41,123 @@ fn impl_type_id(input: &mut DeriveInput) -> TokenStream2 {
     }
 }
 
-mod kw {
-    syn::custom_keyword!(transparent);
+#[derive(Debug, Clone)]
+enum Transparent {
+    NotTransparent,
+    Transparent(Option<syn2::Type>),
+}
+
+impl FromMeta for Transparent {
+    fn from_none() -> Option<Self> {
+        Some(Self::NotTransparent)
+    }
+
+    fn from_word() -> darling::Result<Self> {
+        Ok(Self::Transparent(None))
+    }
+
+    fn from_string(value: &str) -> darling::Result<Self> {
+        let ty = syn2::parse_str(value)?;
+        Ok(Self::Transparent(Some(ty)))
+    }
+}
```

Comment:
Can't this be derived using darling derives?

Reply:
Not with this syntax I believe.. The only things that can be constructed from a word (smth like `#[schema(transparent)]`) are `bool` and `()`, but then there wouldn't be a way to provide an optional explicit type

Reply:
Ok, then how does this code reflects that only `transparent` word can construct `Transparent` struct?

Reply:
It doesn't. In `darling` data model the name of the meta matters only in the `FromAttributes` (and alike). So, to change the kind of word used to create the `Transparent` struct, you change the name if the field in the `SchemaAttributes` struct

```rust
#[derive(Debug, Clone, FromAttributes)]
#[darling(attributes(schema))]
struct SchemaAttributes {
    // change the name here
    transparent: Transparent,
}
```